src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -37,12 +39,14 @@ public partial class BlockProcessor(
     IBlockProcessor.IBlockTransactionsExecutor? blockTransactionsExecutor,
     IWorldState? stateProvider,
     IReceiptStorage? receiptStorage,
-    IBlockhashStore? blockHashStore,
+    ITransactionProcessor transactionProcessor,
```

Comment:
Not sure about directly passing TransactionProcessor:
- until now we are passing concrete types that do something like `IBlockhashStore`, `IBeaconBlockRootHandler`, `IWithdrawalProcessor` and `IConsensusRequestsProcessor` and transaction processor is purposefully hidden under `IBlockProcessor.IBlockTransactionsExecutor` as it can execute differently dependent of the mode (block validation/production/mev production/tracing).
- Here we are only using it to create consensusRequestProcessor if it is not passed, which looks weird.

Reply:
It's the result of merging two different conventions. I think a more elegant solution would be resolving BlockhashStore, BeaconBlockRootHandler, and WithdrawalProcessor in the constructor, but I don't mind if it's done as you suggested.

cc: @rjnrohit

Reply:
If you resolve them in a constructor then they are not composable nor customizable.